PhysicsTools/Heppy/python/physicsobjects/Lepton.py

```diff
@@ -44,6 +44,23 @@ def relIso(self,dBetaFactor=0, allCharged=0):
         rel = self.absIso(dBetaFactor, allCharged)/self.pt()
         return rel
 
+    def absIsoR(self, R=0.3, dBetaFactor=0, allCharged=False):
+        if dBetaFactor>0 and self.puChargedHadronIsoR(R)<0:
+            raise ValueError('If you want to use dbeta corrections, you must make sure that the pu charged hadron iso is available. This should never happen') 
+        neutralIso = self.neutralHadronIsoR(R) + self.photonIsoR(R)
+        #Recover FSR
+        if hasattr(self, 'fsrPhotons'):
+            for gamma in self.fsrPhotons:
+                neutralIso = neutralIso - gamma.pt()
```

Comment:
For FSR, now we implement it in Electron.py and Muon.py since it needs to take into account the vetos in the photon isolation that are different for the different objects.
If you're not using FSR then maybe it's better if you remove the code from here, while if you use FSR then maybe it's better to use the absIsoWithFSR methods (which also supports rhoArea corrections instead of deltaBeta)